app/src/main/java/org/mozilla/fenix/perf/StartupTimeline.kt

```diff
@@ -45,3 +67,29 @@ object StartupTimeline {
         state = StartupTimelineStateMachine.getNextState(state, startingActivity)
     }
 }
+
+/**
+ * A [LifecycleObserver] for [HomeActivity] focused on startup performance measurement.
+ */
+internal class StartupHomeActivityLifecycleObserver(
+    private val frameworkStartMeasurement: StartupFrameworkStartMeasurement,
+    private val startupTimeline: PingType<NoReasonCodes> = Pings.startupTimeline
+) : LifecycleObserver {
+
+    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
+    fun onStop() {
+        GlobalScope.launch { // use background thread due to expensive metrics.
+            // Ensure any last metrics are set before submission.
```

Comment:
Is there a reason why we use GlobalScope? is this the onStop for the Activity, which is only called when the app gets killed?

Reply:
> Is there a reason why we use GlobalScope?

My understanding is that `GlobalScope` has no built in cancellation. Assuming you don't need things to return to the UI thread and don't mind running while the app is backgrounded (e.g. you might want to to conserve battery/CPU), it's fine to use. I think this represents our case. Ideally, I think we'd want to avoid the latter case but, to keep this simple and because it's unlikely to continue running long anyway, I think it's fine.

> is this the onStop for the Activity,

Yes, this observer gets registered on HomeActivity.

> which is only called when the app gets killed?

It should get called in that case and when the app is backgrounded by a user and not killed.

Reply:
So yes, from what I read `GlobalScope` is lost in the void. Is there any event where we would want to cancel this job?

Reply:
Not that I'm aware of – in fact, to keep things simple, my goal is to not to have to deal with the possibility. :)